src/ada/index.ts

```diff
@@ -0,0 +1,35 @@
+import type { ProviderEntity } from '@/entities/provider-entity';
+import type { Account, Address, Balance } from '@/types';
+import { CardanoWallet } from './available-wallets';
+import { connect } from './connect';
+import type { CardanoProviderProps } from './types';
+
+export class CardanoProvider implements ProviderEntity {
+  wallet?: CardanoWallet
+
+  constructor({ wallet }: CardanoProviderProps) {
+    if (wallet)
+      this.wallet = CardanoWallet[wallet.toUpperCase() as keyof typeof CardanoWallet]
+  }
+
+  async connect(): Promise<Account[]> {
+    const accounts = await connect(this.wallet)
+
+    return accounts.map((account, index) => ({
```

Comment:
"The only improvement suggestion I have is to add question marks before the periods (?.) to avoid potential errors in the application. For example, 'accounts?.map'."

Reply:
its not a good practice use `Optional Chaining` in this scenario, the best approach is to validate with `if` block each case of problem. I'm using typescript to ensure the correct return type of the method and im using `error handler pattern` to stop execution before this block line, ensure the correct response.